digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/MunicipalityDao.scala

```diff
@@ -41,6 +41,16 @@ class MunicipalityDao {
       select name_fi from municipality where id = $id""".as[String].first
   }
 
+  def getMunicipalityIdByName(municipalityName: String): List[MunicipalityInfo] = {
+    sql"""
+      select id, ely_nro, name_fi
+      from municipality
+      where LOWER(name_fi) = LOWER($municipalityName)"""
+      .as[(Int, Int, String)].list
+      .map{ case(id, ely, name) =>
+        MunicipalityInfo(id, ely, name)}
+  }
+
```

Comment:
Can we do something a bit more direct like MapViewZoom in both this function and the others that return a MunicipalityInfo?

```
  implicit val getMunicipalityInfo = new GetResult[MunicipalityInfo] {
    def apply(r: PositionedResult) = {
      val id = r.nextInt()
      val ely = r.nextInt()
      val name = r.nextString()

      MunicipalityInfo(id, ely, name)
    }
  }
```

`.as[MunicipalityInfo].list`

Reply:
Another thing is why you are returning a List? There is a possibility of having municipalities with the same name?

Maybe the argument can be changed to List[municipalityName] for reusability for giving more municipalities names.

Reply:
this function was suggested by @MAC-CGI and I tried to mimic the old behaviour

Reply:
Applied first comment